File: python-api/src/api/handlers/polls/getAllPolls/getAllPollsHandler.py
try:
    import unzip_requirements
except ImportError:
    pass

import boto3
import os
import simplejson as json

dynamodb_client = boto3.resource('dynamodb')
TABLE_NAME = os.environ['POLLS_TABLE_NAME']
DYNAMO_POLLS_ID = os.environ['POLLS_ID']

def create_response(code, message):
  return {
    'statusCode': code,
    'headers': {'Content-Type': 'application/json'},
    'body': json.dumps(message)
  }

def getAllPolls(event, context):

    table = dynamodb_client.Table(TABLE_NAME)

    params = {
        'IndexName': 'ListPollsTitles',
        'KeyConditionExpression': 'GSI1PK = :v',
        'ExpressionAttributeValues': {
            ':v': 'POLLS'
        }
    }


    response = table.query(**params)

    print('response: ', response)

    polls = response['Items']

    return create_response(200, polls)

File: nodejs-api/src/api/handlers/polls/getAllPolls/getAllPollsHandler.js
'use strict';

const AWS = require('aws-sdk')
const db = new AWS.DynamoDB.DocumentClient({})

const tableName = process.env.POLLS_TABLE_NAME

const response = (statusCode, message) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(message)
    }
}

const getAllPolls = async () => {
    const params = {
      TableName: tableName,
      IndexName: 'ListPollsTitles',
      KeyConditionExpression: 'GSI1PK = :v',
      ExpressionAttributeValues: {
        ':v': 'POLLS'
      }
    }
  
    try {
      const pollsSearch = await db.query(params).promise()
      const polls = pollsSearch.Items
      return response(200, polls)
    } catch (err) {
      return response(err.statusCode, err)
    }
}

module.exports = {
    getAllPolls
}